src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3772,12 +3779,15 @@ void c_typecheck_baset::typecheck_side_effect_assignment(
     if(o_type0.id() == ID_vector)
     {
       if(
-        o_type1.id() == ID_vector && o_type0.subtype() == o_type1.subtype() &&
-        is_number(o_type0.subtype()))
+        o_type1.id() == ID_vector &&
+        to_vector_type(o_type0).element_type() ==
+          to_vector_type(o_type1).element_type() &&
+        is_number(to_vector_type(o_type0).element_type()))
       {
         return;
       }
-      else if(is_number(o_type0.subtype()) && is_number(o_type1))
+      else if(
+        is_number(to_vector_type(o_type0).element_type()) && is_number(o_type1))
```

Comment:
Perhaps introduce a temporary for `to_vector_type(o_type0)` for all of this?

Reply:
done